ledger/complete/wal/checkpointer.go

```diff
@@ -442,6 +478,10 @@ func StoreCheckpoint(writer io.Writer, tries ...*trie.MTrie) error {
 	// included.
 	for _, t := range tries {
 		rootNode := t.RootNode()
+		if !t.IsEmpty() && rootNode.Height() != ledger.NodeMaxHeight {
+			return fmt.Errorf("height of root node must be %d, but is %d",
+				ledger.NodeMaxHeight, rootNode.Height())
+		}
```

Comment:
A similar check is in V5, I think this would be consistent
https://github.com/onflow/flow-go/blob/feature-checkpoint-v6/ledger/complete/wal/checkpointer.go#L492